lib/wrappers/angular.js

```diff
@@ -27,8 +27,14 @@
             }, angularAdapter);
 
             // Wrap `execute()` in an Angular promise
-            api.execute = function(method, params) {
-                return $q.when(TaggedApi.prototype.execute.call(this, method, params));
+            api.execute = function(method, options) {
+                if (!options.clientId || typeof options.clientId === 'undefined') {
+                    return $q.reject('Client ID is required to make API calls');
+                }
+
+                // append the client ID from the params sent from client
+                TaggedApi.prototype.execute.call(this, options.clientId);
+                return $q.when(TaggedApi.prototype.execute.call(this, method, options.params || {}));
```

Comment:
This is breaking change to the interface for `api.execute`, which is a public method and would require all client consumers to update their code.